### ISSUE TITLE
Immediate crash on Android with latest JSC installed

### ISSUE THREAD

**Issue by Menardi** [react-navigation/react-navigation#5636]
### Current Behavior
After installing the latest jsc, as per [these instructions](https://github.com/react-native-community/jsc-android-buildscripts#how-to-use-it-with-my-react-native-app), the app crashes immediately. The error is the same as described in https://github.com/react-navigation/react-navigation/issues/5350.

```
TypeError: TypeError: undefined is not an object (evaluating 'this._callListeners.bind')

This error is located at:
    in StackViewLayout (at withOrientation.js:30)
    in withOrientation (at StackView.js:79)
    in RCTView (at View.js:45)
    in View (at Transitioner.js:215)
    in Transitioner (at StackView.js:22)
    in StackView (created by Navigator)
    in Navigator (at createKeyboardAwareNavigator.js:12)
    in KeyboardAwareNavigator (at createAppContainer.js:388)
    in NavigationContainer (at renderApplication.js:34)
    in RCTView (at View.js:45)
    in View (at AppContainer.js:98)
    in RCTView (at View.js:45)
    in View (at AppContainer.js:115)
    in AppContainer (at renderApplication.js:33)
```

### Expected Behavior
No crash on startup with the latest jsc.

### How to reproduce
Set up an app using `react-native init`, and add `react-navigation` and `jsc-android` as per their instructions.

Set up the app:
```
react-native init test
cd test
yarn add react-navigation react-native-gesture-handler
react-native link
```

Then modify `MainActivity.java` as per [these instructions](https://reactnavigation.org/docs/en/getting-started.html), and replace `App.js` as described [here](https://reactnavigation.org/docs/en/hello-react-navigation.html#creating-a-stack-navigator).

To install the latest jsc:
```
yarn add jsc-android@236355.1.1
```

Then follow the rest of the installation instructions in the [jsc-android README](https://www.npmjs.com/package/jsc-android#how-to-use-it-with-my-react-native-app). Once you run the app, you should see the error.

### Your Environment

I am testing this on Ubuntu 18.10 with both an Android 9 emulator and Android 7 device.

| software         | version
| ---------------- | -------
| react-navigation | 3.3.2
| react-native     | 0.58.6
| node             | 10.15.1
| npm or yarn      | yarn 1.13.0

**Comment by brentvatne** [comment]
this is unrelated to latest jsc. it's related to whatever babel config you're using and metro cache.

https://github.com/react-navigation/react-navigation/issues/5350
https://github.com/react-navigation/react-navigation/issues/5244

**Comment by Menardi** [comment]
This turned out to be two issues, partially related to the cache, though not a bug with react-navigation, as you said.

For anyone who's facing a similar issue - the project I originally had the problem with uses babel plugins for decorators (`@babel/plugin-proposal-decorators` and `@babel/plugin-proposal-class-properties`). This set up consistently causes the error to occur. So I set up a minimal project, without these decorators, and still got the issue. But that seems to have been caused by some weird Metro caching, despite me clearing as much as I can.

Anyway, a similar issue with decorators is discussed in https://github.com/facebook/react-native/issues/20150. It just happened to manifest itself in this particular set up, though I don't know why. 

The solution to this particular issue is to use the `@babel/plugin-transform-flow-strip-types` plugin (as per https://github.com/facebook/react-native/issues/20150#issuecomment-417858270), configuring it to be invoked before the decorator plugins.
